### PR TITLE
Update API of Docs in module, Time

### PR DESCRIPTION
**Pull request by Maninda**
## Purpose
Update Documentation API of the module, _Time_.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [x] API documentation 
  - [x] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 1]
```diff
@@ -267,11 +343,17 @@ function externSubtractDuration(Time time, int years, int months, int days, int
...
 
 # Change the time-zone of the given time.
+# ```ballerina
+#  string zoneId = "America/Panama";
+#  time:TimeZone zoneValue = {id: zoneId};
+#  time:Time time = {time: 1578488382444, zone: zoneValue};
+#  time:Time|time:Error newTime = time:toTimeZone(time, zoneId);
+# ```
 #
 # + time - The Time record of which the time-zone to be changed
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:353]:
```suggestion
# + time - The Time record of which the time-zone is to be changed
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 2]
```diff
@@ -267,11 +343,17 @@ function externSubtractDuration(Time time, int years, int months, int days, int
...
 # + zoneId - The new time-zone id
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:354]:
```suggestion
# + zoneId - The new time-zone ID
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 3]
```diff
@@ -267,11 +343,17 @@ function externSubtractDuration(Time time, int years, int months, int days, int
...
 # + return - Time object containing time and zone information after the conversion
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:355]:
```suggestion
# + return - Time object containing the time and zone information after the conversion
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 4]
```diff
@@ -267,11 +343,17 @@ function externSubtractDuration(Time time, int years, int months, int days, int
...
-#            or an `time:Error` if failed to format the time
+#            or an error if failed to format the time
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:356]:
```suggestion
#            or else an error if failed to format the time
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 5]
```diff
@@ -282,8 +364,11 @@ function externToTimeZone(Time time, handle zoneId) returns Time|Error = @java:M
 } external;
 
 # Returns the current time value with the system default time-zone.
+# ```ballerina
+#  time:Time now = time:currentTime();
+# ```
 #
-# + return - Time object containing the time and zone information
+# + return - Time object containing the time and the zone information
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:371]:
```suggestion
# + return - Time object containing the time and zone information
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 6]
```diff
@@ -306,6 +394,9 @@ function externNanoTime() returns int = @java:Method {
 } external;
 
 # Returns the Time object correspoding to the given time components and time-zone.
+# ```ballerina
+#  time:Time|error dateTime = time:createTime(2020, 3, 28, 23, 42, 45, 554, "America/Panama");
+# ```
 #
 # + year - The year representation
 # + month - The month-of-year to represent, from 1 (January) to 12 (December)
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:402]:
```suggestion
# + month - The month-of-year to represent from 1 (January) to 12 (December)
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 7]
```diff
@@ -328,10 +419,14 @@ function externCreateTime(int year, int month, int date, int hour, int minute, i
 } external;
 
 # Returns the time for the given string representation based on the given format string.
+# ```ballerina
+#  string timeFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+#  time:Time|error time = time:parse("2020-06-26T09:46:22.444-0500", timeFormat);
+# ```
 #
 # + data - The time text to parse
 # + timeFormat - The format which is used to parse the given text
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:428]:
```suggestion
# + timeFormat - The format, which is used to parse the given text
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 8]
```diff
@@ -328,10 +419,14 @@ function externCreateTime(int year, int month, int date, int hour, int minute, i
...
-# + return - Time object containing time and zone information or an `time:Error` if failed to parse the given string
+# + return - Time object containing time and zone information or an error if failed to parse the given string
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:429]:
```suggestion
# + return - Time object containing the time and zone information or else an error if failed to parse the given string
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal (hunk 5 above)
Comment (Maninda) [reply stdlib/time/src/main/ballerina/src/time/timelib.bal:371]: Zone is a different type of information from time.

File: stdlib/time/src/main/ballerina/src/time/Module.md [hunk 9]
```diff
@@ -1,5 +1,3 @@
-## Module overview
-
 This module provides implementations related to time, date, time zones, and durations. 
 
 The module has two main types as [Time](records/Time.html) and [TimeZone](records/TimeZone.html). The type `Time` represents a time associated with a given time zone. It has `time` and `zone` as attributes. The type `TimeZone` represents the time zone associated with a given time. It has `id` and `offset` as attributes. An `id` can be one of the following:
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/Module.md:3]:
```suggestion
The module has two main types as [Time](records/Time.html) and [TimeZone](records/TimeZone.html). The `Time` type represents a time associated with a given time zone. It has `time` and `zone` as its attributes. The `TimeZone` type represents the time zone associated with a given time. It has `id` and `offset` as its attributes. An `id` can be one of the following:
```

File: stdlib/time/src/main/ballerina/src/time/Module.md [hunk 10]
```diff
@@ -9,148 +7,9 @@ The module has two main types as [Time](records/Time.html) and [TimeZone](record
 * If `id` starts with '+' or '-', the ID is parsed as an offset. Offset can be specified in one of the following ways. +h, +hh, +hh:mm, -hh:mm, +hhmm, -hhmm, +hh:mm:ss, -hh:mm:ss, +hhmmss, -hhmmss
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/Module.md:7]:
```suggestion
* If `id` starts with '+' or '-', the ID is parsed as an offset. The offset can be specified in one of the following ways: +h, +hh, +hh:mm, -hh:mm, +hhmm, -hhmm, +hh:mm:ss, -hh:mm:ss, +hhmmss, -hhmmss
```

File: stdlib/time/src/main/ballerina/src/time/Module.md [hunk 11]
```diff
@@ -9,148 +7,9 @@ The module has two main types as [Time](records/Time.html) and [TimeZone](record
...
 * Also `id` can be a region-based zone ID. The format is '{area}/{city}' eg: "America/Panama". The zones are based on IANA Time Zone Database (TZDB) supplied data.
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/Module.md:8]:
```suggestion
* Also, `id` can be a region-based zone ID. The format is '{area}/{city}' e.g., "America/Panama". The zones are based on IANA Time Zone Database (TZDB) supplied data.
```

File: stdlib/time/src/main/ballerina/src/time/time_errors.bal [hunk 12]
```diff
@@ -14,18 +14,18 @@
...
-# + cause - Any other error, which causes this error.
+# + message - Specific error message of the error
+# + cause - Any other error, which causes this error
 public type Detail record {
     string message;
     error cause?;
 };
 
-# Error reason to specify the time error that occurs in the module
+# Specifies the time error that occurs in the module.
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/time_errors.bal:26]:
```suggestion
# Specifies the time error, which occurs in the module.
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 13]
```diff
@@ -40,7 +40,12 @@ public type Time record {|
...
-# Returns ISO 8601 string representation of the given time.
+# Returns the ISO 8601 string representation of the given time.
+# ```ballerina
+#  time:TimeZone zoneValue = {id: "America/Panama"};
+#  time:Time time = {time: 1578488382444, zone: zoneValue};
+#  string timeString = time:toString(time);
+# ```
 #
 # + time - The Time record to be converted to string
 # + return - The ISO 8601 formatted string of the given time
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:51]:
```suggestion
# + return - The ISO 8601-formatted string of the given time
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 14]
```diff
@@ -58,11 +63,16 @@ function externToString(Time time) returns handle = @java:Method {
...
+#  time:TimeZone zoneValue = {id: "America/Panama"};
+#  time:Time time = {time: 1578488382444, zone: zoneValue};
+#  string|error timeString = time:format(time, time:TIME_FORMAT_RFC_1123);
+# ```
 #
 # + time - The Time record to be formatted
-# + timeFormat - The format which is used to format the time represented by this object
-# + return - The formatted string of the given time or an `time:Error` if failed to format the time
+# + timeFormat - The format, which is used to format the time represented by this object
+# + return - The formatted string of the given time or else an error if failed to format the time
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:75]:
```suggestion
# + return - The formatted string of the given time or else a `time:Error` if failed to format the time
```
need to specify the error type.

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 15]
```diff
@@ -109,8 +129,13 @@ function externGetMonth(Time time) returns int = @java:Method {
...
 # Returns the date representation of the given time.
+# ```ballerina
+#  time:TimeZone zoneValue = {id: "America/Panama"};
+#  time:Time time = {time: 1578488382444, zone: zoneValue};
+#  int day = time:getDay(time);
+# ```
 #
-# + time - The Time record to get the date representation from
+# + time - The Time record to get the date representation
 # + return - The day-of-month, from 1 to 31
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:139]:
```suggestion
# + return - The day-of-month from 1 to 31
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 16]
```diff
@@ -140,8 +170,13 @@ function externGetWeekday(Time time) returns handle = @java:Method {
...
 
 # Returns the hour representation of the given time.
+# ```ballerina
+#  time:TimeZone zoneValue = {id: "America/Panama"};
+#  time:Time time = {time: 1578488382444, zone: zoneValue};
+#  int hour = time:getHour(time);
+# ```
 #
-# + time - The Time record to get the  hour representation from
+# + time - The Time record to get the  hour representation
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:179]:
```suggestion
# + time - The Time record to get the hour representation
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 17]
```diff
@@ -153,8 +188,13 @@ function externGetHour(Time time) returns int = @java:Method {
...
 
 # Returns the minute representation of the given time.
+# ```ballerina
+#  time:TimeZone zoneValue = {id: "America/Panama"};
+#  time:Time time = {time: 1578488382444, zone: zoneValue};
+#  int minute = time:getMinute(time);
+# ```
 #
-# + time - The Time record to get the  minute representation from
+# + time - The Time record to get the  minute representation
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:197]:
```suggestion
# + time - The Time record to get the minute representation
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 18]
```diff
@@ -140,8 +170,13 @@ function externGetWeekday(Time time) returns handle = @java:Method {
...
 # Returns the hour representation of the given time.
+# ```ballerina
+#  time:TimeZone zoneValue = {id: "America/Panama"};
+#  time:Time time = {time: 1578488382444, zone: zoneValue};
+#  int hour = time:getHour(time);
+# ```
 #
-# + time - The Time record to get the  hour representation from
+# + time - The Time record to get the  hour representation
 # + return - The hour-of-day, from 0 to 23
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:180]:
```suggestion
# + return - The hour-of-day from 0 to 23
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 19]
```diff
@@ -153,8 +188,13 @@ function externGetHour(Time time) returns int = @java:Method {
...
 # Returns the minute representation of the given time.
+# ```ballerina
+#  time:TimeZone zoneValue = {id: "America/Panama"};
+#  time:Time time = {time: 1578488382444, zone: zoneValue};
+#  int minute = time:getMinute(time);
+# ```
 #
-# + time - The Time record to get the  minute representation from
+# + time - The Time record to get the  minute representation
 # + return - The minute-of-hour to represent, from 0 to 59
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:198]:
```suggestion
# + return - The minute-of-hour to represent from 0 to 59
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 20]
```diff
@@ -179,8 +224,13 @@ function externGetSecond(Time time) returns int = @java:Method {
...
 # Returns the millisecond representation of the given time.
+# ```ballerina
+#  time:TimeZone zoneValue = {id: "America/Panama"};
+#  time:Time time = {time: 1578488382444, zone: zoneValue};
+#  int milliSecond = time:getMilliSecond(time);
+# ```
 #
-# + time - The Time record to get the millisecond representation from
+# + time - The Time record to get the millisecond representation
 # + return - The milli-of-second, from 0 to 999
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:234]:
```suggestion
# + return - The milli-of-second from 0 to 999
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 21]
```diff
@@ -223,6 +283,14 @@ function externGetTime(Time time) returns [int, int, int, int] = @java:Method {
...
+# ```ballerina
+#  string timeText = "2020-06-26T09:46:22.444-0500";
+#  string timeFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+#  time:Time|error originalTime = time:parse(timeText, timeFormat);
+#  if (originalTime is time:Time) {
+#      time:Time newTime = time:addDuration(originalTime, 1, 1, 1, 1, 1, 1, 1);
+#  }
+# ```
 #
 # + time - The Time record to add the duration to
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:295]:
```suggestion
# + time - The Time record to add the duration 
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 22]
```diff
@@ -267,11 +343,17 @@ function externSubtractDuration(Time time, int years, int months, int days, int
...
+# ```
 #
-# + time - The Time record of which the time-zone to be changed
-# + zoneId - The new time-zone id
-# + return - Time object containing time and zone information after the conversion
-#            or an `time:Error` if failed to format the time
+# + time - The Time record of which the time-zone is to be changed
+# + zoneId - The new time-zone ID
+# + return - Time object containing the time and zone information after the conversion
+#            or else an error if failed to format the time
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:356]:
```suggestion
#            or else  a `time:Error` if failed to format the time
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 23]
```diff
@@ -306,9 +394,12 @@ function externNanoTime() returns int = @java:Method {
...
 
 # Returns the Time object correspoding to the given time components and time-zone.
+# ```ballerina
+#  time:Time|error dateTime = time:createTime(2020, 3, 28, 23, 42, 45, 554, "America/Panama");
+# ```
 #
 # + year - The year representation
-# + month - The month-of-year to represent, from 1 (January) to 12 (December)
+# + month - The month-of-year to represent from 1 (January) to 12 (December)
 # + date - The day-of-month to represent, from 1 to 31
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:403]:
```suggestion
# + date - The day-of-month to represent from 1 to 31
```

Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:403]:
```suggestion
# + date - The day-of-month to represent from 1 to 31
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 24]
```diff
@@ -306,9 +394,12 @@ function externNanoTime() returns int = @java:Method {
...
 # + hour - The hour-of-day to represent, from 0 to 23
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:404]:
```suggestion
# + hour - The hour-of-day to represent from 0 to 23
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 25]
```diff
@@ -306,9 +394,12 @@ function externNanoTime() returns int = @java:Method {
...
 # + minute - The minute-of-hour to represent, from 0 to 59
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:405]:
```suggestion
# + minute - The minute-of-hour to represent from 0 to 59
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 26]
```diff
@@ -328,10 +419,14 @@ function externCreateTime(int year, int month, int date, int hour, int minute, i
...
+# ```ballerina
+#  string timeFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+#  time:Time|error time = time:parse("2020-06-26T09:46:22.444-0500", timeFormat);
+# ```
 #
 # + data - The time text to parse
-# + timeFormat - The format which is used to parse the given text
-# + return - Time object containing time and zone information or an `time:Error` if failed to parse the given string
+# + timeFormat - The format, which is used to parse the given text
+# + return - Time object containing the time and zone information or else an error if failed to parse the given string
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:429]:
```suggestion
# + return - Time object containing the time and zone information or else  a `time:Error` if failed to parse the given string
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 27]
```diff
@@ -166,8 +206,13 @@ function externGetMinute(Time time) returns int = @java:Method {
...
 # Returns the second representation of the given time.
+# ```ballerina
+#  time:TimeZone zoneValue = {id: "America/Panama"};
+#  time:Time time = {time: 1578488382444, zone: zoneValue};
+#  int second = time:getSecond(time);
+# ```
 #
-# + time - The Time record to get the second representation from
+# + time - The Time record to get the second representation
 # + return - The second-of-minute, from 0 to 59
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:216]:
```suggestion
# + return - The second-of-minute from 0 to 59
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal (hunk 23 above)
Comment (Maninda) [reply stdlib/time/src/main/ballerina/src/time/timelib.bal]: Already added.

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 28]
```diff
@@ -267,11 +343,17 @@ function externSubtractDuration(Time time, int years, int months, int days, int
...
+# ```
 #
-# + time - The Time record of which the time-zone to be changed
-# + zoneId - The new time-zone id
-# + return - Time object containing time and zone information after the conversion
-#            or an `time:Error` if failed to format the time
+# + time - The Time record of which the time-zone is to be changed
+# + zoneId - The new time-zone ID
+# + return - Time object containing the time and zone information after the conversion
+#            or else  a `time:Error` if failed to format the time
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:356]:
```suggestion
#            or else a `time:Error` if failed to format the time
```

File: stdlib/time/src/main/ballerina/src/time/timelib.bal [hunk 29]
```diff
@@ -282,8 +364,11 @@ function externToTimeZone(Time time, handle zoneId) returns Time|Error = @java:M
 } external;
 
 # Returns the current time value with the system default time-zone.
```
Comment (praneesha) [inline stdlib/time/src/main/ballerina/src/time/timelib.bal:366]:
```suggestion
# Returns the current time value with the default system time-zone.
```

**Pull request merged**